js/glowingbear.js

```diff
@@ -656,6 +657,39 @@ weechat.controller('WeechatCtrl', ['$rootScope', '$scope', '$store', '$timeout',
         window.requestAnimationFrame(scroll);
     };
 
+    $scope.parseHost = function() {
+        //The host field is multi purpose for advanced users
+        //There can be a combination of host, port and path
+        //If host is specified here the dedicated port field is disabled
+
+        var parts;
+
+        //host
+        var regexHost = /^([^:\/]*|\[.*\])$/;
+        var regexHostPort = /^([^:]*|\[.*\]):(\d+)$/;
+        var regexHostPortPath = /^([^:]*|\[.*\]):(\d*)\/(.+)$/;
+        if((parts = regexHost.exec(settings.host)) !== null)
+        {
+            settings.hostOnly = parts[1];
```

Comment:
`hostOnly` is not a great name; should probably rename the old `host` to `hostField` (and bind the input to that) and use `host` for the new one. That would require handling the case where `host` is set but `hostField` is not upon page load, but I think the increased clarity is worth it.

Reply:
I agree and I had thought about it but it seemed too invasive to touch the saved settings. Now I changed the names to the more logical ones like you suggested but had to add a bit of code for when users upgrade or they'll lose their settings. This extra bit of code should probably dissapear in the far future (as if).